lib/std/mem.zig

```diff
@@ -10,16 +10,67 @@ const testing = std.testing;
 const Endian = std.builtin.Endian;
 const native_endian = builtin.cpu.arch.endian();
 
+pub const PageSizeCheck = enum {
+    /// Does nothing if the page size is invalid
+    unsafe,
+    /// Throws an error if the page size is invalid
+    throw,
+    /// Asserts if the page size is invalid
+    assert,
+    /// Panics if the page size is invalid
+    panic,
+
+    pub fn makeType(comptime self: PageSizeCheck) type {
+        return switch (self) {
+            .assert, .unsafe, .panic => usize,
+            .throw => error{InvalidPageSize}!usize,
+        };
+    }
+};
```

Comment:
This is way too abstract. I don't know what problem you're trying to solve but this is not the solution.

Reply:
Was a solution to [this concern](https://github.com/ziglang/zig/pull/17382#discussion_r1344631378). I can just remove the check and make it up to the caller to check if the size is valid.